src/resmom/linux/alps.c

```diff
@@ -4696,20 +4984,23 @@ alps_request_parent(int fdin)
  * stdout to be read by the parent.
  *
  * @param[in] msg XML message to send to ALPS client
- *
+ * @param[in] basil_ver BASIL version indicates context i.e. did control
+ *	      arrive at this function as a result of processing the
+ *	      Inventory Query or System Query.
  * @return pointer to filled in response structure
  * @retval NULL no result
  *
  */
 static basil_response_t *
-alps_request(char *msg)
+alps_request(char *msg, char *basil_ver)
 {
 	int toChild[2];
 	int fromChild[2];
-	int status;
+	int status = 0;
 	pid_t pid;
 	pid_t exited;
-	size_t msglen, wlen = -1;
+	size_t msglen = 0;
+	size_t wlen = -1;
```

Comment:
wlen is used to capture the return of fwrite. Which either returns zero in case of failure or value greater than zero.
Therefore initialize value is set to -1, to make sure that initialized value is not reflecting error value before it is used. If it's ok i would like to keept it as it is.

Reply:
That's fine with me.  Thanks for the explanation.